### ISSUE TITLE
Error: trying to use more than locked

### ISSUE THREAD

**Issue by arthurwolf** [c9s/bbgo#398]
The error:

`',panic: executeTrade exception, wanted to use more than the locked balance: trying to use more than locked: locked 0.000000 < want to use 1428.571429\n\ngoroutine 58 [running]:\ngithub.com/c9s/bbgo/pkg/backtest.(*SimplePriceMatching).executeTrade(0xc0005deb40, {0x0, 0x9, 0x7, {0x11fb113, 0x8}, 0x3fba6d43246cf1c6, 0x40cb07619616824d, 0x4096524924922bb3, {0xc0002f7f00, ...}, ...})\n\t/home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211223162419-49e516b80e13/pkg/backtest/matching.go:190 +0x453\ngithub.com/c9s/bbgo/pkg/backtest.(*SimplePriceMatching).SellToPrice(0xc0005deb40, 0x9d5170)\n\t/home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211223162419-49e516b80e13/pkg/backtest/matching.go:381 +0x54c\ngithub.com/c9s/bbgo/pkg/backtest.(*SimplePriceMatching).processKLine(0xc0005deb40, {0x92571, {0xc000263400, 0x7}, {0xc000263410, 0x7}, {0x0, 0xed5add214, 0x0}, {0x0, ...}, ...})\n\t/home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211223162419-49e516b80e13/pkg/backtest/matching.go:420 +0x559\ngithub.com/c9s/bbgo/pkg/backtest.(*Stream).Connect.func2()\n\t/home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211223162419-49e516b80e13/pkg/backtest/stream.go:90 +0x25e\ncreated by github.com/c9s/bbgo/pkg/backtest.(*Stream).Connect\n\t/home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211223162419-49e516b80e13/pkg/backtest/stream.go:77 +0x6b8\n,'`

Config:

```
arthur@crypto:~$ cat /home/arthur/tmp/bbgo-config/3daed2d450ebb89d6841fb0d5d577f9c.yaml
---
sessions:
  binance:
    exchange: binance
    envVarPrefix: binance
backtest:
  startTime: "2020-01-13"
  endTime: "2020-01-15"
  symbols:
  - ONGUSDT
  account:
    makerCommission: 15 # TODO: Set the right values for binance, figure out what the number means
    takerCommission: 15
    balances:
      ONG: 0.0
      USDT: 10000
exchangeStrategies:
- on: binance
  grid:
    symbol: ONGUSDT
    # was: quantity: 1428.57142857142866887443
    amount: 1428.57142857142866887443
    gridNumber: 7.00000000000000000000
    profitSpread: 0.00034697142857142901
    upperPrice: 0.10504000000000000836
    lowerPrice: 0.10292800000000000560
    long: true
arthur@crypto:~$ 
```

Command: 

`cd /home/arthur/bbgo/ ;timeout --kill-after=630s 600s /home/arthur/bbgo/bbgo backtest --exchange binance --dotenv /home/arthur/tmp/bbgo-env/bbgo-env-ONGUSDT.yaml --config /home/arthur/tmp/bbgo-config/3daed2d450ebb89d6841fb0d5d577f9c.yaml --base-asset-baseline --output /home/arthur/tmp/bbgo-cache/3daed2d450ebb89d6841fb0d5d577f9c/ 2>&1`

**Comment by arthurwolf** [comment]
Other instance of this error:

![Screenshot from 2021-12-24 10-52-45](https://user-images.githubusercontent.com/108821/147342755-797587cd-c0e5-4f30-a2a8-705e0e1220f8.png)

**Comment by c9s** [comment]
this is the concurrent issue,

should also be fixed in 442afe8eb9baefab437a3245823af3aade89e315
